virttest/test_setup.py

```diff
@@ -1339,7 +1341,7 @@ def set_vf(self, pci_pf, vf_no="0"):
         :params vf_no: Number of Vfs to be virtualized
         :return: True on success, False on failure
         """
-        cmd = "echo %s > /sys/bus/pci/devices/%s/sriov_numvfs" % (vf_no, pci_pf)
+        cmd = "echo %s > /sys/class/net/%s/device/sriov_numvfs" % (vf_no, pci_pf)
```

Comment:
yes,  this updated set_vf() can work with get_pf_mlx().  i have checked the whole file , get_pf_mlx() only used for get pf_devices when execute set_vf().  So I  prefer to use get_pf_ids()  instead of  get_pf_mlx(),  no need to change set_vf().

Reply:
We use get_pf_mlx() to get pf_devices not only when set_vf  but also when set ip for pf. You can refer to L1500, which sets ip for the interface. With updated set_vf, get_pf_mlx can work well for both.

Reply:
Yes, it can avoid the error. I will update, thanks!

Reply:
Yes, it can avoid the error if lshw is not installed. But it will call get_pf_ids() to get the id first, which would be less efficient . I'm not sure which way is better.

Reply:
I think , "lshw" is not a package avocado depend on, it is only used for sr-iov test.  From my understanding, avocado dependency is something to support the framework.  So, i prefer to change another method to get the nic interface.
@vivianQizhu   what do you think about this problem ?

Reply:
@yanan-fu Agree with you, I don't prefer to add a not very necessary dependency if have other soft way work around.

Reply:
Thanks @yanan-fu @vivianQizhu , updated, please help review again.